internal/util/http.go

```diff
@@ -0,0 +1,174 @@
+package util
+
+/*
+ Copyright 2017 - 2021 Crunchy Data Solutions, Inc.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/crunchydata/postgres-operator/internal/logging"
+)
+
+const (
+	clientHeader = "X-Client-Update-State"
+)
+
+var (
+	client  HTTPClient
+	backoff = wait.Backoff{
+		Duration: 1 * time.Second,
+		Factor:   float64(2),
+		Steps:    4,
+		Cap:      30 * time.Second,
+	}
+
+	// TODO(benjaminjb): Get real URL
+	updateCheckURL    = "http://localhost:8080"
+	updateCheckPeriod = 24 * time.Hour
+)
+
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+func init() {
+	// Since we create this client once during startup,
+	// we want each connection to be fresh, hence the non-default transport
+	// See https://github.com/golang/go/issues/43905 and https://github.com/golang/go/issues/23427
+	// for discussion of problems with long-lived connections
+	client = &http.Client{
+		Timeout: 5 * time.Second,
+		Transport: &http.Transport{
+			DisableKeepAlives:   true,
+			MaxIdleConnsPerHost: -1,
```

Comment:
❓ I don't see negative `MaxIdleConnsPerHost` documented in `net/http` or mentioned in the linked issues. (Perhaps I missed it?) Do we have something that ensures it works the way we expect? Do we need another link that shows it?

Reply:
Originally I was going off a pattern I had seen elsewhere for clean/new http connections that set both, but after reading the Transport code more, it seems like DisableKeepAlives provides the behavior we want (that is, given the if statement `if t.DisableKeepAlives || t.MaxIdleConnsPerHost < 0 {` we don't need both (and there's no other place where a greater-than-zero MaxIdleConnsPerHost interferes with DisableKeepAlives.